src/WebJobs.Script/Host/BlobLeaseManager.cs

```diff
@@ -0,0 +1,173 @@
+
+        public static async Task<BlobLeaseManager> CreateAsync(string accountConnectionString, TimeSpan leaseTimeout, string hostId, TraceWriter traceWriter)
+        {
+            if (leaseTimeout.TotalSeconds < 15 || leaseTimeout.TotalSeconds > 60)
+            {
+                throw new ArgumentOutOfRangeException(nameof(leaseTimeout), $"The {nameof(leaseTimeout)} should be between 15 and 60 seconds");
+            }
+
+            ICloudBlob blob = await GetLockBlobAsync(accountConnectionString);
+            var manager = new BlobLeaseManager(blob, leaseTimeout, hostId, traceWriter);
+            return manager;
+        }
+
+        public static BlobLeaseManager Create(string accountConnectionString, TimeSpan leaseTimeout, string hostId, TraceWriter traceWriter)
+        {
+            return CreateAsync(accountConnectionString, leaseTimeout, hostId, traceWriter).GetAwaiter().GetResult();
+        }
+
+        private async Task AcquireOrRenewLeaseAsync()
+        {
+            try
+            {
+                if (HasLease)
+                {
+                    await _lockBlob.RenewLeaseAsync(new AccessCondition { LeaseId = LeaseId });
+                    _traceWriter.Verbose("Host lock lease renewed.");
+                }
+                else
+                {
+                    LeaseId = await _lockBlob.AcquireLeaseAsync(_leaseTimeout, _hostId);
+                    _traceWriter.Info($"Host lock lease acquired by host ID '{_hostId}'.");
+
+                    // We've successfully acquired the lease, change the timer to use our renewal interval
+                    _timer.Change(_renewalInterval, _renewalInterval);
+                }
+            }
+            catch (StorageException exc) when (exc.RequestInformation.HttpStatusCode == 409)
+            {
+                if (HasLease)
+                {
+                    LeaseId = null;
+                    _timer.Change(_leaseRetryInterval, _leaseRetryInterval);
+
+                    _traceWriter.Info("Failed to renew host lock lease.");
+                }
+                else
+                {
+                    _traceWriter.Verbose($"Host '{_hostId}' failed to acquire host lock lease. Another host has an active lease.");
+                }
+            }
+        }
+
+        private static async Task<ICloudBlob> GetLockBlobAsync(string accountConnectionString)
+        {
+            CloudStorageAccount account = CloudStorageAccount.Parse(accountConnectionString);
+            CloudBlobClient client = account.CreateCloudBlobClient();
+
+            var container = client.GetContainerReference(HostContainerName);
+
+            await container.CreateIfNotExistsAsync();
+            CloudBlockBlob blob = container.GetBlockBlobReference(LockBlobName);
+            if (!await blob.ExistsAsync())
+            {
+                await blob.UploadFromStreamAsync(new MemoryStream());
+            }
```

Comment:
I'm assuming you're talking about line 132 here. Yes, there is a chance. I'll address this.